### ISSUE TITLE
java.lang.ClassCastException: ClusterTopologyChangedEventVisitor cannot be cast to ConnectedEventVisitor

### ISSUE THREAD

**Issue by CharlesG-Branch** [dropwizard/dropwizard-redis#13]
I'm using this version of the module:
```
        <groupId>io.dropwizard.modules</groupId>
        <artifactId>dropwizard-redis</artifactId>
        <version>1.3.12-1</version>
```

& the issue appears when following the https://github.com/dropwizard/dropwizard-redis#basic-lettuce-client from the readme.

From looking at the code I think the issue is happening here:
```
eventWrapper -> eventVisitors.forEach(eventWrapper::accept)
```
The implementations of `VisitableEventWrapper` are limiting what visitors can be passed to `accept` because of the template, but the `forEach` above is applying visitors of all types. I'm surprised the compiler didn't catch this.

I think this problem can be resolved by just removing the template from `VisitableEventWrapper`. I can put up a PR and test out later today if you'd like.

Full stacktrace:
```
Exception in thread "my-redis-use-case-eventExecutorLoop-1-1" reactor.core.Exceptions$ErrorCallbackNotImplemented: java.lang.ClassCastException: io.dropwizard.redis.metrics.event.visitor.ClusterTopologyChangedEventVisitor cannot be cast to io.dropwizard.redis.metrics.event.visitor.ConnectedEventVisitor
Caused by: java.lang.ClassCastException: io.dropwizard.redis.metrics.event.visitor.ClusterTopologyChangedEventVisitor cannot be cast to io.dropwizard.redis.metrics.event.visitor.ConnectedEventVisitor
	at io.dropwizard.redis.metrics.event.wrapper.VisitableConnectedEventWrapper.accept(VisitableConnectedEventWrapper.java:8)
	at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
	at io.dropwizard.redis.RedisClientFactory.lambda$null$1(RedisClientFactory.java:74)
	at java.util.Optional.ifPresent(Optional.java:159)
	at io.dropwizard.redis.RedisClientFactory.lambda$build$2(RedisClientFactory.java:74)
	at reactor.core.publisher.LambdaSubscriber.onNext(LambdaSubscriber.java:130)
	at reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.runAsync(FluxPublishOn.java:398)
	at reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.run(FluxPublishOn.java:484)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
	at io.netty.util.concurrent.PromiseTask.run(PromiseTask.java:73)
	at io.netty.util.concurrent.DefaultEventExecutor.run(DefaultEventExecutor.java:66)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:905)
	at com.codahale.metrics.InstrumentedThreadFactory$InstrumentedRunnable.run(InstrumentedThreadFactory.java:66)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by mzamani1** [comment]
Thanks for the report! If you’d like to submit a PR, that would be great! Otherwise, I’m happy to fix if not.

**Comment by mzamani1** [comment]
Thanks again @CharlesG-Branch! I just released `1.4.0` containing this fix, among other dependency version bumps.
